cmd/kubeadm/app/util/arguments.go

```diff
@@ -30,26 +30,22 @@ import (
 func BuildArgumentListFromMap(baseArguments map[string]string, overrideArguments map[string]string) []string {
 	var command []string
 	var keys []string
-	for k := range overrideArguments {
-		keys = append(keys, k)
-	}
-	sort.Strings(keys)
-	for _, k := range keys {
-		v := overrideArguments[k]
-		// values of "" are allowed as well
-		command = append(command, fmt.Sprintf("--%s=%s", k, v))
+
+	for k, v := range baseArguments {
+		if _, overrideExists := overrideArguments[k]; !overrideExists {
+			overrideArguments[k] = v
```

Comment:
this still has the issue that was reported before: content of arguments in outer scope of that function will be changed if we do assignment to `baseArguments` or `overrideArguments`

Reply:
@jiaxuanzhou don't modify any of the arguments. Just copy `baseArguments` into another `map[string]string` and then override that copy with the contents of `overrideArguments`. Finally, fetch its keys, sort them and construct the command line options slice to be returned.

Reply:
thx, i got you and modified.